src/Provider/GenericProvider.php

```diff
@@ -141,7 +141,8 @@ protected function getRequiredOptions()
     private function assertRequiredOptions(array $options)
     {
         $missing = array_diff_key(array_flip($this->getRequiredOptions()), $options);
-        if ($missing) {
+
+        if (!empty($missing)) {
```

Comment:
> The two statements are effectively equivalent.

Yup, effectively equivalent, but not expressively.

> It should be fine to leave as is, and we avoid a function call, if you want an optimization argument.

Sure thing. Weak, but valid. ;)